controllers/authenticate-token.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const _ =  require('lodash'),
+      auth = require('basic-auth'),
+      jwt = require('jsonwebtoken'),
+      path = require('path');
+
+const models = require(path.resolve('./models')),
+      config = require(path.resolve('./config'));
+
+const jwtSecret = config.jwt.secret;
+const jwtExpirationTime = config.jwt.expirationTime;
+
+async function generateTokenBehavior(data) {
+  let username, password, token, authenticated, verified, user;
+  const responseData = {};
+  // checks for authorisation headers
+  // and take username and password from header
+  if (_.has(data, 'headers.authorization')) {
+    try{
+      username = auth(data).name;
+      password = auth(data).pass;
+    } catch(e) {
+      // TODO error status
+      responseData.status = 401;
+      responseData.message = 'Invalid authorization header';
+      responseData.data = {};
+      return responseData;
+    }
+  } else {
+    responseData.status = 401;
+    responseData.message = 'Invalid authorization header';
+    responseData.data = {};
+    return responseData;
+  }
+
+  try{
+    // checks if user exists
+    const userExists = await models.user.exists(username);
+    if (!userExists || username === undefined || password === undefined) {
+      responseData.status = 401;
+      responseData.message = 'User doesn\'t exist or lack of username or password';
```

Comment:
For security reasons all 401 messages should be the same, not to reveal any information to an attacker. https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Correct_Response_Example